source/data-modeling/field-types.txt

```diff
@@ -0,0 +1,558 @@
+
+When you declare a field as a ``Regexp``, {+odm+} converts Ruby regular
+expressions to BSON regular expressions when storing the result into your
+database. The database returns the field as a ``Bson::Regexp::Raw`` instance.
+You can use the ``compile`` method on ``BSON::Regexp::Raw`` instances to convert
+the data back to a Ruby regular expression.
+
+The following example creates a ``Token`` class and specifies the ``pattern``
+field as a ``Regexp``:
+
+.. literalinclude:: /includes/data-modeling/field-types.rb
+   :language: ruby
+   :start-after: # start-regexp
+   :end-before: # end-regexp
+
+.. important::
+
+   Converting a BSON regular expression to a Ruby regular expression might
+   produce a different regular expression than the original. This difference is
+   due to the differences between the syntax Onigmo and PCRE.
+
+BigDecimal
+~~~~~~~~~~
+
+You can use the ``BigDecimal`` type to store numbers with increased precision.
+{+odm+} stores ``BigDecimal`` values in two different ways, depending on the
+value of ``Mongoid.map_big_decimal_to_decimal128``:
+
+- If set to ``True``, {+odm+} stores ``BigDecimal`` values as BSON ``Decimal128``
+  values.
+- If set to ``False`` (default), {+odm+} stores ``BigDecimal`` values as a string.
+
+Consider the following limitations when setting the
+``Mongoid.map_big_decimal_to_decimal128`` option to true:
+
+- ``Decimal128`` has a limited range and precision.
+  ``Decimal128`` has a maximum value of approximately ``10^6145`` and a minimum
+  of approximately ``-10^6145``, with a maximum of 34 bits of precision. If you
+  are storing values that are outside of these limits, we recommend storing them
+  as a string instead.
+- ``Decimal128`` accepts signed ``NaN`` values, but ``BigDecimal`` does not.
+  Retrieving signed ``NaN`` ``Decimal128`` values from the database as
+  ``BigDecimal`` returns the value unsigned.
+- ``Decimal128`` maintains trailing zeroes, but ``BidDecimal`` does not.
+  Because of this, retrieving ``Decimal128`` values from the database as
+  ``BigDecimal`` might result in a loss of precision.
+
+.. note::
+   
+   When you set the ``Mongoid.map_big_decimal_to_decimal128`` option to ``false``
+   and store a ``BigDecimal`` into an untyped field, you cannot query the field 
+   as a ``BigDecimal``. Because the value is stored as a string, querying 
+   the untyped field for a ``BigDecimal`` value does not find the value in the
+   database. To find the value, you must first convert the query value to a string.
+
+   You can avoid this issue by specifying the field as a ``BigDecimal`` type,
+   rather than untyped.
+
+Specify Fields with Strings or Symbols
+---------------------------------------
+
+You can use strings or symbols to specify certain field types in {+odm+}, instead of
+using their class names. The following example specifies the ``state`` field by
+using the class name, a string, and a symbol:
+
+.. code-block:: ruby
+
+   class Order
```

Comment:
S: change to address or something more intuitive. it also is confusing to put state as an integer

Reply:
Updated "state" to be more intuitive